atomos.framework/src/main/java/org/atomos/framework/base/AtomosCommands.java

```diff
@@ -86,29 +92,57 @@ private String getState(Bundle b) {
 		}
 	}
 
-	public void install(String name, String loaderType, File moduleDir) throws BundleException {
+	@Descriptor("Install a new layer")
+	public void install(@Descriptor("Name of the layer") String name,
+			@Descriptor("LoaderType of the layer [OSGI, SINGLE, MANY]") String loaderType,
+			@Descriptor("Directory where the AtomosBundles are stored") File moduleDir)
+					throws BundleException {
+
 		if (!moduleDir.isDirectory()) {
 			System.out.println("The specified path is not a directory: " + moduleDir.getAbsolutePath());
+			return;
+		}
+
+		final Optional<LoaderType> oLoaderType = Stream.of(LoaderType.values())
+				.filter(e -> e.name().equalsIgnoreCase(loaderType))
+				.findAny();
+
+		if (!oLoaderType.isPresent()) {
+			final String v = Stream.of(LoaderType.values())
+					.map(LoaderType::name)
+					.collect(Collectors.joining(", "));
+			System.out.printf("The specified loaderType is not valid. Use one of %s", v);
+			return;
 		}
 
-		AtomosLayer layer = runtime.addLayer(Collections.singletonList(runtime.getBootLayer()), name, LoaderType.valueOf(loaderType), moduleDir.toPath());
-		
-		List<Bundle> bundles = new ArrayList<>();
-		for (AtomosBundleInfo atomosBundle : layer.getAtomosBundles()) {
+		final AtomosLayer layer = runtime.addLayer(Collections.singletonList(runtime.getBootLayer()),
+				name, oLoaderType.get(), moduleDir.toPath());
+
+		final List<Bundle> bundles = new ArrayList<>();
+		for (final AtomosBundleInfo atomosBundle : layer.getAtomosBundles()) {
 			bundles.add(atomosBundle.install(null));
 		}
-		for (Bundle b : bundles) {
+		for (final Bundle b : bundles) {
 			b.start();
 		}
 		layers(layer, new HashSet<>());
 	}
 
-	public void uninstall(long id) throws BundleException {
-		AtomosLayer layer = runtime.getById(id);
+
+	@Descriptor("Uninstall the layer with the given id")
+	public void uninstall(@Descriptor("Id of the layer") long id) throws BundleException {
+
+		final AtomosLayer layer = runtime.getById(id);
 		if (layer == null) {
 			System.out.println("No Atomos Layer with ID: " + id);
 		} else {
-			layer.uninstall();
+			try {
+				layer.uninstall();
+				System.out.printf("Sucessfully uninstalled Atomos Layer \"%s\" with ID: %s",
+						layer.getName(), id);
+			} catch (final Exception e) {
```

Comment:
Why catch and rethrow here?  The try block doesn't seem necessary.

Reply:
I want to sysout a failing message.

Reply:
OK fine, but I do think your original PR didn't include the systout :)